modules/login/php/passwordexpiry.class.inc

```diff
@@ -102,6 +103,14 @@ class PasswordExpiry extends \NDB_Form
             return array('password' => 'You cannot keep the same password');
         }
 
+        // Ensure that the password is not the user's email
+        // Otherwise an exception is thrown in User.class::updatePassword
+        // TODO This validation should be done on the front-end too.
+        if ($plaintext == $data['Email']) {
+            $this->tpl_data['error_message'] = "Password cannot be user's email";
+            return array('password' => "Password cannot be user's email");
```

Comment:
The error message displayed to the user should be in the second person, rather than calling them "user". ie. "Can not user your email as your password" (Though maybe someone has a better suggestion for phrasing.)

Reply:
```suggestion
            return array('password' => "You cannot use your email as your password.);
```